codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/XmlShapeSerVisitor.java

```diff
@@ -106,7 +106,7 @@ protected void serializeMap(GenerationContext context, MapShape shape) {
             writer.insertTrailingNewline();
 
             // Serialize zero values as empty values.
-            GoValueAccessUtils.writeIfZeroValue(context.getModel(), writer, shape.getValue(), "v[i]", () -> {
+            GoValueAccessUtils.writeIfZeroValue(context.getModel(), writer, shape.getValue(), "v[key]", () -> {
```

Comment:
This makes me wonder if `queryParamsMap` serialization need something similar.  What is this changing exactly?

Reply:
`i` was not a valid variable here, looks like this was some sort of copy paste error, but there was no code-gen that exercised this path before. the variable in scope that it should have been using was `key`. Probably was copied from the code that handled lists.